projects/helper/chain/provenance.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Token Mapping for Provenance. Note that
+ * YLDS is pending on CoinGecko, but is a stablecoin.
+ * Additionally, LRWA and REIT are also 1:1 with USD
+ */
+const tokenMapping = {
+    hash: "hash-2",
+    "eth.figure.se": "ethereum",
+    "usd.trading": "usd-coin",
+    "usdc.figure.se": "usd-coin",
+    // "ylds.fcc": "ylds",
+    "usdt.figure.se": "tether",
+    "xrp.figure.se": "ripple",
+    "sol.figure.se": "solana",
+    "btc.figure.se": "bitcoin",
+    "uni.figure.se": "uniswap",
+    "lrwa.figure.markets": "usd-coin",
+    "link.figure.se": "chainlink",
+    'reit.figure.markets': "usd-coin",
+};
+
+/**
+ * Denoms are listed as their base unit. For example, HASH is returned as
+ * nhash, and needs to be converted to it's CoinGecko Value.
+ * We do this by querying the denom metadata and using the provided
+ * exponent and base unit to convert.
+ * @param tokenObject An object with key: token name (in base units) 
+ * and value: amount of tokens
+ * Example: {
+ *   nhash: 100000000
+ * }
+ */
+const convertToCoinGeckoApiId = async (tokenObject) =>
+    await Promise.all(
+        Object.keys(tokenObject).map(async (t) => {
+            // Get the denom exponent information
+            const response = await fetch(
```

Comment:
no, this is really bad, this will fetch denom metadata each time,  use this structure instead: https://github.com/DefiLlama/DefiLlama-Adapters/blob/main/projects/helper/tokenMapping.js#L41


then you can do `api.add(token, bal)` in the adapter code and add 

`return sumTokens2({ api })`, it will take care of transforming to cg tokens

Reply:
This is now resolved.